umbraco/Umbraco-CMS
# Netcore: Replace IUmbracoRequestLifetime with EventAggregator nofitications

Fixes [AB#9928](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/9928) 

### Description
- Replaces the `IUmbracoRequestLifetime` with `EventAggregator` notifications
- Had issues with profiling when handled by `EventAggregator` notifications, due to `MiniProfiler.Current` was null, if set in an async notification.
  - Therefore web profiling is handled explicitly in the `UmbracoRequestMiddleware`. This also ensures the profiling is enabled and disabled as the first and last thing, respectively.
- Increased the SQL lock timeout. We still need to make this configurable but for now, the larger timespan makes the acceptance tests more stable.

### Test
- Ensure the events are still executed.
- Ensure profiling works
  - Visit https://localhost:44331/?umbDebug=true
    - First request will include the boot profiling
    - Second only the request profiling
